app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
@davidh-raybeam
1. Soft delete preserves performance statistics.
2. I take it you meant sites, and yes.
3. No

Regarding David's sample code, the first loop is all you need. Both `forget_proxy` and`disable_proxy` queue proxy decommissions when the proxy is used by no more sites.  Also, use `disable_proxy` instead. It preserves performance data.